### DISCUSSION TITLE
Relate States with Timezones

### DISCUSSION THREAD

**Discussion by NadeemAraman** [dr5hn/countries-states-cities-database#451] [Ideas]
Dear dr5hn , first of all thank you for this wonderful api , second is there anyway that i can bring up or get the time zone of a specific state/city ? i'm saying this because for example russia has more than 8 different timezones and the project i'm working on need to relate each state to its timezone , thanks in advance

**Comment by dr5hn** [comment]
@NadeemAraman This is not implemented at the moment. But will surely implement this in near future.

**Comment by mogspls** [comment]
Hi! Is there any update on this? Would like to hear your thoughts on it. Thanks!

**Comment by mpathy** [comment]
There is the Timezone API, the free plan is sufficient to add those fields:
https://developers.google.com/maps/documentation/timezone/overview?hl=de

Also this might even work without subscription, just be sure you dont do too much requests.
But if one uses it with long pauses in between you can just do with the demo username:
http://www.geonames.org/export/web-services.html#timezone

You can do requests to this form field: https://timezonedb.com/
Or this: https://www.zeitverschiebung.net/en/city/2192362

Also there is a NodeJS library:
https://github.com/kevinroberts/city-timezones

An SQL file here with instructions to check on updates:
https://timezonedb.com/download

Also a project that looks up timezone by the boundaries with data from OpenStreetmap:
https://github.com/evansiroky/timezone-boundary-builder

A more elaborate list than here:
https://stackoverflow.com/questions/16086962/how-to-get-a-time-zone-from-a-location-using-latitude-and-longitude-coordinates/16086964#16086964

They either work with long / lat - values already provided in the data or the city names.

I also would like to have the timezones added already in this repository.
Then the only thing I need to do for my project is import the sql and I am done :)

In the meantime, I add it with scripts but it would be so much easier if included already.

**Comment by sami-eljabali** [comment]
In looking into this myself. It seems at least in the USA placing timezones at the state level is not accurate enough. Neither at the city level is it accurate enough, instead necessitates to place timezones at the postal code level. Why? There are several cities in the United States that have different time zones within the same state or across state lines. Here are a few examples:

* Kansas City: Kansas City, Missouri, and Kansas City, Kansas, are neighboring cities divided by the state line. Kansas City, Missouri, follows Central Time (CT), while Kansas City, Kansas, follows Central Time but observes Daylight Saving Time (CT/CDT). This means that during Daylight Saving Time, there is a one-hour time difference between the two cities.

* Texarkana: Texarkana is a city that is split between Texas and Arkansas. The Texas side of Texarkana follows Central Time (CT), while the Arkansas side follows Eastern Time (ET). This results in a one-hour time difference between the two sides of the city.

* South Bend: South Bend, Indiana, is located near the border with Michigan. Most of Indiana follows Eastern Time (ET), but some counties in the northwest, including St. Joseph County where South Bend is located, observe Central Time (CT). This means that South Bend, Indiana, is on Eastern Time (ET), while its neighboring cities in Michigan, such as Niles or Benton Harbor, are on Central Time (CT).

* Bristol: Bristol is a city that straddles the border between Tennessee and Virginia. The Tennessee side of Bristol follows Eastern Time (ET), while the Virginia side follows Eastern Time but observes Daylight Saving Time (ET/EDT). This creates a one-hour time difference between the two sides of the city during Daylight Saving Time.

These are just a few examples of cities with different time zones within or across state lines. There may be other instances where cities near time zone boundaries have different time zones.
